src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy

```diff
@@ -72,6 +91,33 @@ class PalantirDockerPlugin implements Plugin<Project> {
                 workingDir dockerDir
                 commandLine 'docker', 'push', ext.name
             }
+
+            dockerfileZip.with {
+                from(ext.resolvedDockerfile)
+            }
+
+            // Configure docker-compose templating
+            if (ext.resolvedDockerComposeTemplate.exists()) {
+                def dockerDependencies = project.configurations.docker.resolvedConfiguration.resolvedArtifacts
+                def templateTokens = dockerDependencies.collectEntries {
+                    def version = it.moduleVersion.id
+                    [("{{${version.group}:${version.name}}}"): version.version]
+                }
+
+                generateDockerCompose.with {
+                    from(ext.resolvedDockerComposeTemplate)
+                    into(ext.resolvedDockerComposeFile.parentFile)
+                    rename { fileName -> fileName.replace(
+                        ext.resolvedDockerComposeTemplate.name, ext.resolvedDockerComposeFile.name) }
```

Comment:
it's a little different from Dockerfile: the former is optional but must exist if the user specifies it, the latter is required in either case.